runtime/ms_rest_azure/spec/polling_state_spec.rb

```diff
@@ -92,6 +98,35 @@ module MsRestAzure
 
       expect(polling_state.get_delay).to eq(5)
     end
+
+    it 'should return succeeded status for a 200 response and no provisioning status' do
+      response = double('response', :status => 200, :headers =>
+          { 'Azure-AsyncOperation' => 'async_operation_header',
+            'Location' => 'location_header'})
+      azure_response = double('response',
+                              :request => nil,
+                              :response => response,
+                              :body => nil)
+
+      polling_state = PollingState.new azure_response, 0
+
+      expect(polling_state.status).to eq(AsyncOperationStatus::SUCCESS_STATUS)
+    end
+
+    it 'should return in progress status for a 201 response and no provisioning status' do
+      response = double('response', :status => 201, :headers =>
+          { 'Azure-AsyncOperation' => 'async_operation_header',
+            'Location' => 'location_header'})
+      azure_response = double('response',
+                              :request => nil,
+                              :response => response,
+                              :body => nil)
+
+      polling_state = PollingState.new azure_response, 0
+
+      expect(polling_state.status).to eq(AsyncOperationStatus::IN_PROGRESS_STATUS)
```

Comment:
Discussed with @devigned These services have created before the arm guidelines. That was the reason .NET SDK and Node SDK follow the same logic and with this PR we are aligning our code with them